playwright/_impl/_connection.py

```diff
@@ -609,7 +609,21 @@ def _extract_stack_trace_information_from_stack(
 
 
 def _filter_none(d: Mapping) -> Dict:
-    return {k: v for k, v in d.items() if v is not None}
+    result = {}
+    for k, v in d.items():
+        if v is None:
+            continue
+        elif isinstance(v, dict):
+            filtered_v = _filter_none(v)
+            if filtered_v:
+                result[k] = filtered_v
+        elif isinstance(v, Mapping):
```

Comment:
Do we have cases for this branch? looks the same so we can combine the two branches into one if statement?